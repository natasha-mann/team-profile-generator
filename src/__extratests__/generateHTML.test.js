const {
  generateHTML,
  generateManagerCard,
  generateEngineerCard,
  generateInternCard,
} = require("../utils/generateHTML");

const inquirer = require("inquirer");

jest.mock("../utils/generateHTML.js");
const inquirerSpy = jest.spyOn(inquirer, "prompt");

describe("generateHTML", () => {
  const mockAnswers = (answers = {}) => {
    inquirerSpy.mockResolvedValue(answers);
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test("should return the expected HTML markup", () => {
    const mockData = {
      teamName: "international",
      manager: {
        name: "test",
        employeeID: "test",
        email: "test",
        officeNumber: "test",
      },
      teamMember: {
        name: "test",
        employeeID: "test",
        email: "test",
        github: "test",
        school: "test",
      },
    };

    mockAnswers(mockData);

    const actual = generateHTML();

    expect(actual).toMatchSnapshot();
  });
});

describe("generateManagerCard", () => {
  const mockAnswers = (answers = {}) => {
    inquirerSpy.mockResolvedValue(answers);
  };

  test("should return the expected HTML markup", () => {
    const mockData = {
      manager: {
        name: "test",
        employeeID: "test",
        email: "test",
        officeNumber: "test",
      },
    };

    mockAnswers(mockData);

    const actual = generateManagerCard(mockData);

    expect(actual).toMatchSnapshot();
  });
});

describe("generateEngineerCard", () => {
  const mockAnswers = (answers = {}) => {
    inquirerSpy.mockResolvedValue(answers);
  };

  test("should return the expected HTML markup", () => {
    const mockData = {
      teamMember: {
        name: "test",
        employeeID: "test",
        email: "test",
        github: "test",
        school: "test",
      },
    };

    mockAnswers(mockData);

    const actual = generateEngineerCard(mockData);

    expect(actual).toMatchSnapshot();
  });
});

describe("generateInternCard", () => {
  const mockAnswers = (answers = {}) => {
    inquirerSpy.mockResolvedValue(answers);
  };

  test("should return the expected HTML markup", () => {
    const mockData = {
      teamMember: {
        name: "test",
        employeeID: "test",
        email: "test",
        github: "test",
        school: "test",
      },
    };

    mockAnswers(mockData);

    const actual = generateInternCard(mockData);

    expect(actual).toMatchSnapshot();
  });
});
