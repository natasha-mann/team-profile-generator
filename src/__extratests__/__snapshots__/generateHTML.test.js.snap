// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateEngineerCard should return the expected HTML markup 1`] = `undefined`;

exports[`generateHTML should return the expected HTML markup 1`] = `undefined`;

exports[`generateInternCard should return the expected HTML markup 1`] = `undefined`;

exports[`generateManagerCard should return the expected HTML markup 1`] = `undefined`;
